grammars/css.cson

```diff
@@ -20,294 +20,1205 @@
+    'patterns': [
+      {
+        'include': '#feature-query-operators'
+      }
+      {
+        'include': '#feature-query'
+      }
+    ]
+  'feature-query-operators':
+    'patterns': [
+      {
+        'match': '(?i)(?<=[\\s()]|^|\\*/)(and|not|or)(?=[\\s()]|$|/\\*)'
+        'name': 'keyword.operator.logical.$1.feature.css'
+      }
+      {
+        'include': '#rule-list-innards'
+      }
+    ]
+  'font-features':
+    'begin': '''(?xi)
+      ((@)(annotation|character-variant|ornaments|styleset|stylistic|swash))
+      (?=[\\s{@'";]|/\\*|$)\\s*
+    '''
+    'beginCaptures':
       '1':
+        'name': 'keyword.control.at-rule.${3:/downcase}.css'
+      '2':
         'name': 'punctuation.definition.keyword.css'
-    'end': '\\s*((?=;|\\}))'
-    'name': 'meta.at-rule.import.css'
+    'end': '(?<=})\\s*'
+    'endCaptures':
+      '0':
+        'name': 'punctuation.terminator.rule.css'
+    'name': 'meta.at-rule.${3:/downcase}.css'
     'patterns': [
       {
-        'include': '#string-double'
+        'begin': '{'
+        'beginCaptures':
+          '0':
+            'name': 'punctuation.section.property-list.begin.css'
+        'end': '(})|(?<=})'
+        'endCaptures':
+          '0':
+            'name': 'punctuation.section.property-list.end.css'
+        'name': 'meta.property-list.font-feature.css'
+        'patterns': [
+          {
+            # Font-feature name
+            'match': '''(?x)
+              (?: [-a-zA-Z_]    | [^\\x00-\\x7F] )   # First letter
+              (?: [-a-zA-Z0-9_] | [^\\x00-\\x7F] )*  # Remainder of identifier
+            '''
+            'name': 'variable.font-feature.css'
+          }
+          {
+            'include': '#rule-list-innards'
+          }
+        ]
+      }
+    ]
+  'functions':
+    'patterns': [
+      # Calculation
+      {
+        'begin': '(?i)(?<![\\w-])(calc)(\\()'
+        'beginCaptures':
+          '1':
+            'name': 'support.function.calc.css'
+          '2':
+            'name': 'punctuation.section.function.begin.bracket.round.css'
+        'end': '\\)'
+        'endCaptures':
+          '0':
+            'name': 'punctuation.section.function.end.bracket.round.css'
+        'name': 'meta.function.calc.css'
+        'patterns': [
+          {
+            'match': '[*/]|(?<=\\s|^)[-+](?=\\s|$)'
+            'name': 'keyword.operator.arithmetic.css'
+          }
+          {
+            'include': '#property-values'
+          }
+        ]
       }
+      # Colours
       {
-        'include': '#string-single'
+        'begin': '(?i)(?<![\\w-])(rgba?|hsla?)(\\()'
+        'beginCaptures':
+          '1':
+            'name': 'support.function.misc.css'
+          '2':
+            'name': 'punctuation.section.function.begin.bracket.round.css'
+        'end': '\\)'
+        'endCaptures':
+          '0':
+            'name': 'punctuation.section.function.end.bracket.round.css'
+        'name': 'meta.function.color.${1:/downcase}-value.css'
```

Comment:
Is the differentiation between rgb(a) and hsl(a) really needed?  Would `meta.function.color.css` suffice?  I don't want to go too overboard on overly-specific selectors.